# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: workflows/validation-linux.yml
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/.github/workflows/validation-linux.yml b/third_party/rust/naga/.github/workflows/validation-linux.yml
new file mode 100644
--- /dev/null
+++ b/third_party/rust/naga/.github/workflows/validation-linux.yml
@@ -0,0 +1,21 @@
+name: validation-linux
+on:
+  pull_request:
+    paths:
+    - 'tests/out/spv/*.spvasm'
+    - 'tests/out/glsl/*.glsl'
+    - 'tests/out/dot/*.dot'
+    - 'tests/out/wgsl/*.wgsl'
+
+jobs:
+  validate-linux:
+    name: SPIR-V + GLSL
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install tools
+        run: sudo apt-get install spirv-tools glslang-tools graphviz
+      - run: make validate-spv
+      - run: make validate-glsl
+      - run: make validate-dot
+      - run: make validate-wgsl